iaso/api/forms.py

```diff
@@ -431,44 +423,81 @@ def destroy(self, request, *args, **kwargs):
         log_modification(original, destroyed_form, FORM_API, user=request.user)
         return response
 
-    @action(detail=True, methods=["get"], permission_classes=[HasFormPermissionOrSignedURL])
+    @action(detail=True, methods=["get"])
     def manifest(self, request, *args, **kwargs):
         """Returns a xml manifest file in the openrosa format for the Form
 
-        This is used for the mobile app and Enketo to fetch the list of file attached to the Form
+        This is used for the mobile app to fetch the list of files attached to the Form
         see https://docs.getodk.org/openrosa-form-list/#the-manifest-document
         """
         form = self.get_object()
-        attachments = form.attachments.all()
-        media_files = []
-        for attachment in attachments:
-            attachment_file_url: str = attachment.file.url
-            if not attachment_file_url.startswith("http"):
-                # Needed for local dev
-                attachment_file_url = public_url_for_enketo(request, attachment_file_url)
-
-            media_files.append(
-                f"""<mediaFile>
-                        <filename>{escape(attachment.name)}</filename>
-                        <hash>md5:{attachment.md5}</hash>
-                        <downloadUrl>{escape(attachment_file_url)}</downloadUrl>
-                    </mediaFile>"""
-            )
+        return generate_manifest_for_form(form, request)
+
+    @action(detail=True, methods=["get"], permission_classes=[ReadOnly])
+    def manifest_enketo(self, request, pk, *args, **kwargs):
+        """Returns a xml manifest file in the openrosa format for the Form
+
+        This is used by enketo to fetch the list of files attached to the Form
+        see https://docs.getodk.org/openrosa-form-list/#the-manifest-document
+        """
+        enketo_secret = enketo_settings("ENKETO_SIGNING_SECRET")
+        if verify_signed_url(request, enketo_secret):
+            app_id = request.query_params.get(APP_ID, None)
```

Comment:
Should we really allow `app_id=None` here?

If the response is no, I'd go for something like that:

```python
app_id = AppIdSerializer(data=request.query_params).get_app_id(raise_exception=True)
```